cmd/root.go

```
@@ -2,57 +2,93 @@ package cmd
 
-		if len(files) == 0 {
-			fmt.Println("✅ No staged files to review.")
-			return
+	for _, reviewer := range reviewers {
+		if reviewer == nil {
+			return nil, fmt.Errorf("reviewer list contains a nil value")
 		}
+	}
+
+	newSource, err := source.New(env)
+	if err != nil {
+		return nil, err
+	}
+
+	return &comm{
+		env:       env,
+		reviewers: reviewers,
+		source:    newSource,
+	}, nil
+}
 
-		hadIssues := false
+type CommandInterpreter interface {
+	Execute() error
+}
 
-		for _, file := range files {
-			content, err := os.ReadFile(file)
-			if err != nil {
-				fmt.Printf("⚠️ Could not read %s: %v\n", file, err)
-				continue
-			}
+type comm struct {
+	env       env.EnvironmentManager
+	reviewers []review.Reviewer
+	source    source.Source
+}
 
-			fmt.Printf("🔍 Reviewing %s...\n", file)
-			result := review.AnalyzeCode(string(content), file)
-			format.PrintToTerminal(file, result)
+func (c *comm) Execute() error {
+	files, err := c.source.GetFiles()
+	if err != nil {
+		return fmt.Errorf("failed to get files form git: %w", err)
+	}
 
-			// crude check to see if we should fail for Git hook
-			if review.ContainsCritical(result) {
-				hadIssues = true
-			}
+	if len(files) == 0 {
+		return nil
+	}
+
+	for _, file := range files {
+		if !c.hasExt(file) {
+			continue
 		}
 
-		if hadIssues {
-			fmt.Println("❌ Critical issues found. Commit aborted.")
-			os.Exit(1)
+		err := c.executeReview(file)
+		if err != nil {
+			return fmt.Errorf("execute, could not read file: %w", err)
 		}
-	},
+	}
+
+	return c.generateReportSummary()
+}
+
+func (c *comm) hasExt(fileName string) bool {
+	return c.env.ShouldProcessFile(fileName)
 }
 
-func Execute() {
-	cobra.CheckErr(rootCmd.Execute())
+func (c *comm) executeReview(fileName string) error {
+	for _, reviewer := range c.reviewers {
+		fmt.Printf("Reviewing %s...\n", fileName)
+		if err := reviewer.AnalyzeCode(fileName); err != nil {
+			return fmt.Errorf("failed to analyze file: %w", err)
+		}
+	}
+
+	return nil
```

The generateReportSummary method has the same issue as Execute - if one reviewer's summary fails, the summaries for subsequent reviewers won't be 
generated.
General suggestions:
1. Consider using a more descriptive name than comm for the struct that implements CommandInterpreter.
2. Add context to error messages to make debugging easier.
3. Consider implementing a more robust error handling strategy that collects errors rather than failing fast.
4. Add documentation comments for exported types and functions.
5. Consider adding logging instead of direct fmt.Printf calls for better control over output.
6. Consider adding unit tests to verify the behavior of the code.
The code is generally well-structured with good separation of concerns, but could benefit from more robust error handling and clearer naming.